app/Listeners/QueueMessage.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Gladiator\Listeners;
+
+use Mail;
+use Gladiator\Events\QueueMessageRequest;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class QueueMessage implements ShouldQueue
+{
+    public $mail;
+
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct(Mail $mail)
+    {
+        $this->mail = $mail;
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  QueueMessageRequest  $event
+     * @return void
+     */
+    public function handle(QueueMessageRequest $event)
+    {
+        $content = $event->message;
+
+        $type = $content->type;
+
+        Mail::send('messages.' . $type, ['content' => $content], function ($msg) use ($content) {
```

Comment:
You probs don't need the `Mail` facade if you have it included in the constructor. Can you do:

```
$this->mail->send(...)
```